lib/gui/components/update-notifier/services/update-notifier.js

```diff
@@ -16,10 +16,24 @@
 
 'use strict';
 
+const _ = require('lodash');
 const semver = require('semver');
 const etcherLatestVersion = require('etcher-latest-version');
 
-module.exports = function($http, $q, ModalService, UPDATE_NOTIFIER_SLEEP_TIME, ManifestBindService, SettingsModel) {
+module.exports = function($http, $q, ErrorService, ModalService, UPDATE_NOTIFIER_SLEEP_TIME, ManifestBindService, SettingsModel) {
+
+  // Memoize the version Promise
+  const memoizeVersion = (func) => {
```

Comment:
Yeah, I think `_.memoize` *should* work fine with Promises, but I don't know for sure.